### PR TITLE
[shared_preferences] add a no-op android folder to web/macos packages

### PR DESCRIPTION
**Pull request by amirh**
## Description

Adding an android/ folder as a temporary mitigation for: https://github.com/flutter/flutter/issues/46898

### CODE REVIEW

File: packages/shared_preferences/shared_preferences_macos/android/gradle.properties [hunk 1]
```diff
@@ -0,0 +1,4 @@
+org.gradle.jvmargs=-Xmx1536M
+android.enableR8=true
+android.useAndroidX=true
+android.enableJetifier=true
```
Comment (mklim) [inline packages/shared_preferences/shared_preferences_macos/android/gradle.properties]: These should be removed.

File: packages/shared_preferences/shared_preferences_macos/android/build.gradle [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+}
+
+apply plugin: 'com.android.library'
+
+android {
+    compileSdkVersion 28
+
+    defaultConfig {
+        minSdkVersion 16
+        testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
```
Comment (mklim) [inline packages/shared_preferences/shared_preferences_macos/android/build.gradle]: nit: This should probably be removed.

File: packages/shared_preferences/shared_preferences_web/android/gradle.properties [hunk 3]
```diff
@@ -0,0 +1,4 @@
+org.gradle.jvmargs=-Xmx1536M
+android.enableR8=true
+android.useAndroidX=true
+android.enableJetifier=true
```
Comment (mklim) [inline packages/shared_preferences/shared_preferences_web/android/gradle.properties]: These should also probably be removed.

File: packages/shared_preferences/shared_preferences_web/android/build.gradle [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+}
+
+apply plugin: 'com.android.library'
+
+android {
+    compileSdkVersion 28
+
+    defaultConfig {
+        minSdkVersion 16
+        testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
```
Comment (mklim) [inline packages/shared_preferences/shared_preferences_web/android/build.gradle]: nit: This should probably be removed.

File: packages/shared_preferences/shared_preferences_macos/android/gradle.properties (hunk 1 above)
Comment (amirh) [reply packages/shared_preferences/shared_preferences_macos/android/gradle.properties]: done

File: packages/shared_preferences/shared_preferences_macos/android/build.gradle (hunk 2 above)
Comment (amirh) [reply packages/shared_preferences/shared_preferences_macos/android/build.gradle]: done

File: packages/shared_preferences/shared_preferences_web/android/gradle.properties (hunk 3 above)
Comment (amirh) [reply packages/shared_preferences/shared_preferences_web/android/gradle.properties]: done

File: packages/shared_preferences/shared_preferences_web/android/build.gradle (hunk 4 above)
Comment (amirh) [reply packages/shared_preferences/shared_preferences_web/android/build.gradle]: done

**Comment by amirh** [comment]
@mklim  PTAL

File: packages/shared_preferences/shared_preferences_macos/android/gradle.properties (hunk 1 above)
Comment (mklim) [reply packages/shared_preferences/shared_preferences_macos/android/gradle.properties]: Maybe this still needs to be pushed? I'm still seeing these lines in a refreshed https://github.com/flutter/plugins/pull/2409/files.

Comment (mklim) [reply packages/shared_preferences/shared_preferences_macos/android/gradle.properties]: Never mind, got the changes now.

**Pull request merged**